### DISCUSSION TITLE
Category and Books model

### DISCUSSION THREAD

**Discussion by LOGESH-B** [gbowne1/codebooker#130] [Ideas]
How the category and books are mapped
* Wheather user provide the category/
* Wheather user selects from the Provided category 
* or something else like finding the category by ISBN number is it  so then the ISBN api?

Note:On the current books model there is no category attribute

**Comment by gbowne1** [comment]
I would provide the user with both ways however we might have to implement one method at a time for the user to select the categories. I had originally thought about some sort of selector in the ui to pick the categories and subcategories.  You will have books that as an example of this it will fit the web development category but could potentially be subcategories by some other subject or category, like a framework or language.

The model and its schema can be fixed

We can then work out other methods of entering books to the database like from a scanner or phone.

**Reply by LOGESH-B** [reply]
Can you tell me how can I proceed with adding new books #128

**Reply by gbowne1** [reply]
From the UI? @LOGESH-B

**Reply by LOGESH-B** [reply]
There is no category in the Books model so can i prooced without category for now or can i make dropdown with some category

**Reply by gbowne1** [reply]
I missed adding category to the model by accident.  

The UI has a Category column and then in the left drawer menu / sidenav.. is a separate filter which you can use as a picker and should be used to filter the table content.  You would click on the tabs in the sidenav and it would filter out all the stuff  that isnt in that category say like Web Development.  

there are table sorting arrows in the table header.  

It would probably be easier if you show me with a screenshot what you are talking about adding. @LOGESH-B

**Reply by LOGESH-B** [reply]
My doubts are about how user going to select the category while adding new book (issue #128 )
* Can we provide the pre defined category(Dropdown)
* Or anything else you are planned to find the category of the book

**Reply by LOGESH-B** [reply]
Because we are showing the Category in the side panel
So if we let the user to type the category means it may be a problem 

If we are implementing Dropdown to select the category from the list,then we need to provide categories list which will cover all types of books

**Reply by gbowne1** [reply]
I thought it might be easier to select categories in the sidebar/sidenav/drawer.  What would define the category would be whatever the category is listed as in the table. in order to enter a new book there is a modal that pops up and it has a category field and that should enter all of that data in the database.

**Reply by LOGESH-B** [reply]
You are saying that we are going to provide the list of category during entering new book to user in category field
am I correct?
is that so, what will user do when the new book not comes under any of the category that listed?

**Reply by gbowne1** [reply]
yes the add new book modal has that category field.  If you click on the + at the bottom left of the table it should show the modal.  I have found that that modal isn't adding any value typed into the category field on to the table.  technically that field on the add book should populate with a drop down with the pre-defined categories we have already however, I think if someone either missed definining a new category they could provide it by clicking on the category in that row/column on whatever book had missing data and just fill in the data.

**Reply by gbowne1** [reply]
@BlackBond06
